tests/test_engine.py

```diff
@@ -17,112 +17,106 @@
 
 
 def test_number():
-    assert Number(1).evaluate({Number(0): {ADDR1}}) == False
-    assert Number(1).evaluate({Number(1): {ADDR1}}) == True
-    assert Number(1).evaluate({Number(2): {ADDR1, ADDR2}}) == False
+    assert Number(1).evaluate({Number(0): {ADDR1}}) is False
+    assert Number(1).evaluate({Number(1): {ADDR1}}) is True
+    assert Number(1).evaluate({Number(2): {ADDR1, ADDR2}}) is False
```

Comment:
i think [evaluate](https://github.com/mandiant/capa/blob/8fe88f601f7d5c6fa775830c44f74147c68d4ba7/capa/engine.py#L56) returns a `Result` instance that implements [`__bool__`](https://github.com/mandiant/capa/blob/8fe88f601f7d5c6fa775830c44f74147c68d4ba7/capa/features/common.py#L96), which makes it easy to compare with True/False, like `if expr.evaluate(...): ...` but I'm not sure its correct to use `is`, because this compares specific object instances.

to make this more clear, perhaps we should update these tests to wrap the calls to `evaluate` with a cast to bool, like `bool(expr.evaluate(...)) is True`.

it was probably a mistake to implement `evaluate` and `Result` like this because the behavior is not obvious.